src/core/server/logging/logging_service.mock.ts

```diff
@@ -45,7 +45,7 @@ const createLoggingServiceMock = () => {
     context,
     ...mockLog,
   }));
-  mocked.asLoggerFactory.mockImplementation(() => createLoggingServiceMock());
+  mocked.asLoggerFactory.mockImplementation(() => mocked);
```

Comment:
This was causing to be unable to collect logs in integration tests, as the actual logger used in the whole server goes throu this call

https://github.com/elastic/kibana/blob/033193b52a0bebb350067f8ca09ecded85d757f7/src/core/server/root/index.ts#L42-L44

`asLoggerFactory` was returning a new, non-accessible instance from tests, even with proper mocking.

Reply:
it theory it could create a shared state between 2 consecutive calls. Let see if we hit once

Reply:
Two calls to `createLoggingServiceMock` are currently returning two distincts `MockedLogger`. The shared state was achieved by mocking `src/core/server/logging/logging_service.ts` to always returns the same mocked instance. But even in that case, `logger.asLoggerFactory()` was creating a distinct one. (Or maybe I did not understand what you meant)

Reply:
```js
test('shares state', () => {
  const loggerParent = loggingServiceMock.create();
  const loggerA = loggerParent.asLoggerFactory();
  const loggerB = loggerParent.asLoggerFactory();

  expect(loggerA).toBe(loggerB);

  const loggerAA = loggerA.get('aa');
  loggerAA.info('hi, AA');

  const loggerBB = loggerA.get('bb');
  loggerBB.info('hi, BB');

  expect(loggingServiceMock.collect(loggerA).info).toMatchInlineSnapshot(`
    Array [
      Array [
        "hi, AA",
      ],
      Array [
        "hi, BB",
      ],
    ]
  `);
});
```

Reply:
Exactly. This test passes on my branch, and fails on master.